Unable to open app in browser — backend crashes and API unreachable

I tried to run the app locally and here’s exactly what I did and observed:

From the repo root I ran npm run dev and got: npm ERR! Missing script: "dev".
I then ran from the backend folder:
cd backend
npm run dev
Nodemon started and [server.js](vscode-file://vscode-app/c:/Users/Sarveshwari/AppData/Local/Programs/Microsoft%20VS%20Code/8761a5560c/resources/app/out/vs/code/electron-browser/workbench/workbench.html) printed Server started on http://localhost:5000, but the process immediately crashed with a MySQL error from mysql2:
Error: Access denied for user ''@'localhost' (using password: YES) (coming from [connect.js](vscode-file://vscode-app/c:/Users/Sarveshwari/AppData/Local/Programs/Microsoft%20VS%20Code/8761a5560c/resources/app/out/vs/code/electron-browser/workbench/workbench.html)).
The server repeatedly attempted to restart and crashed each time.
I started the frontend:
cd frontend
npm run dev
Vite served the frontend at http://localhost:5173/, but the UI could not load data because the backend API was down/unreachable.
I checked [.env](vscode-file://vscode-app/c:/Users/Sarveshwari/AppData/Local/Programs/Microsoft%20VS%20Code/8761a5560c/resources/app/out/vs/code/electron-browser/workbench/workbench.html) and it contained placeholders (no real MySQL credentials), so the DB connection was failing.
I also noticed the backend CORS was configured for http://localhost:5000, while the frontend runs on http://localhost:5173, which would prevent cross-origin requests when the backend is up.
Observed error messages and files involved:

Missing script: "dev" — [package.json](vscode-file://vscode-app/c:/Users/Sarveshwari/AppData/Local/Programs/Microsoft%20VS%20Code/8761a5560c/resources/app/out/vs/code/electron-browser/workbench/workbench.html)
Access denied for user ''@'localhost' — [connect.js](vscode-file://vscode-app/c:/Users/Sarveshwari/AppData/Local/Programs/Microsoft%20VS%20Code/8761a5560c/resources/app/out/vs/code/electron-browser/workbench/workbench.html) / missing values in [.env](vscode-file://vscode-app/c:/Users/Sarveshwari/AppData/Local/Programs/Microsoft%20VS%20Code/8761a5560c/resources/app/out/vs/code/electron-browser/workbench/workbench.html)
Potential CORS mismatch — [server.js](vscode-file://vscode-app/c:/Users/Sarveshwari/AppData/Local/Programs/Microsoft%20VS%20Code/8761a5560c/resources/app/out/vs/code/electron-browser/workbench/workbench.html) vs frontend at http://localhost:5173
Result: frontend loads, but I cannot use the app in the browser because the backend crashes on startup (DB auth failure) and the API is thus unavailable.

--- a/backend/connect.js
+++ b/backend/connect.js
@@ -14,7 +14,10 @@ export const db=mysql.createConnection(
 
 db.connect(
     (err)=>{
-        if(err) throw err;
+        if(err){
+            console.error("MySQL connection error:", err.message || err);
+            return;
+        }
         console.log("Connected to MySQL Server");
     }
 )

